libs/platform/src/lib/components/form/datetime-picker/datetime-picker.component.ts

```diff
@@ -0,0 +1,345 @@
+    markWeekends = true;
+
+    /** @Input Whether to show spinner buttons */
+    @Input()
+    spinnerButtons = true;
+
+    /**
+     * Whether user wants to show week numbers next to days
+     */
+    @Input()
+    showWeekNumbers = true;
+
+    /** Whether or not to show the datetime picker footer with submit/cancel buttons. */
+    @Input()
+    showFooter = true;
+
+    /** Event emitted when the state of the isOpen property changes. */
+    @Output()
+    isOpenChange = new EventEmitter<boolean>();
+
+    /** Event thrown every time calendar active view is changed */
+    @Output()
+    public readonly activeViewChange: EventEmitter<FdCalendarView> = new EventEmitter<FdCalendarView>();
+
+    /** Event emitted when the date changes. This can be a time or day change. */
+    @Output()
+    datetimeChange: EventEmitter<FdDatetime> = new EventEmitter<FdDatetime>();
+
+    /** Event emitted when popover closes. */
+    @Output()
+    readonly onClose: EventEmitter<void> = new EventEmitter<void>();
+
+    @ViewChild(DatetimePickerComponent)
+    dateTimePickerComponent: DatetimePickerComponent;
+
+    /** Whether the time component should be meridian (am/pm). */
+    /** @hidden */
+    private _meridian: boolean;
+
+    /**
+     *  The state of the form control - applies css classes.
+     *  Can be `success`, `error`, `warning`, `information` or blank for default.
+     */
+    private _state: Status;
+
+    @ViewChild(DatetimePickerComponent, { static: true, read: ElementRef })
+    protected _elRef: ElementRef;
+
+    /**
+     * Function used to disable certain dates in the calendar.
+     * @param fdDate FdDate
+     */
+    @Input()
+    disableFunction = function (fdDate: FdDate): boolean {
+        return false;
+    };
+
+    constructor(
+        protected _cd: ChangeDetectorRef,
+        readonly elementRef: ElementRef,
+        @Optional() @Self() readonly ngControl: NgControl,
+        @Optional() @Self() readonly ngForm: NgForm,
+        @Optional() @SkipSelf() @Host() formField: FormField,
+        @Optional() @SkipSelf() @Host() formControl: FormFieldControl<any>
+    ) {
+        super(_cd, ngControl, ngForm, formField, formControl);
+    }
+    /** @hidden */
+    ngOnInit(): void {
+        super.ngOnInit();
+        // set default placeholder value
+        if (!this.placeholder) {
+            this.placeholder = this.format;
+        }
+    }
+    /** @hidden */
+    ngAfterViewInit(): void {
+        super.ngAfterViewInit();
+    }
+    /** @hidden */
+    ngOnDestroy(): void {
+        super.ngOnDestroy();
+        this.dateTimePickerComponent?.ngOnDestroy();
+    }
+    /** @hidden */
+    writeValue(value: FdDatetime): void {
+        super.writeValue(value);
+        // emit events
+        if (value) {
+            this._handleDatetimeChange(value);
+        }
+    }
+
+    /**
+     * @hidden
+     * override base functionality to catch new disabled state
+     */
+    setDisabledState(disabled: boolean): void {
```

Comment:
should it bewith prefix `_`?

Reply:
this method overrides BaseInput's `setDisabledState`, so prefixing would make this a separate method right?